### DISCUSSION TITLE
🚀 Fluid Field individual block/group visibility control

### DISCUSSION THREAD

**Discussion by pyrobob83** [ExpressionEngine/ExpressionEngine#3792] [Feature Requests]
### Suggest an idea for ExpressionEngine

**Is your feature request related to a problem? Please describe.**
Right now there's no way to disable individual fields or field groups within a Fluid Field, meaning there's no way start building a new section and save it (perhaps for review/approval by someone else) without publishing it to the page.

**Describe the solution you'd like**
In it's simplest form, I'd like some way to mark individual fields / field groups within a Fluid field so they don't get rendered, but are still part of the entry.  But the most flexible and most useful way I could think of to implement this would be to allow each individual field/group to be conditionally displayed based on role... that way you could select 'none' to completely disable the field or only select SuperAdmin to disable for most, but allow frontend review for logged in SuperAdmins.  Then you could also configure pages to display different content to different audiences, all built out via a regular Fluid Field.

If the role-based visibility is feasible, from a UI perspective, if a field/group is restricted, there should be some kind of visual indicator for what the visibility is in the field/group heading - perhaps the roles with visibility just listed in parentheses after the field/group name if specific visibility is set (ie anything other than "all").

What do we think? Is this a good idea?  Is it feasible?

**Comment by shbchk** [comment]
Try this approach:
1. Create a new Toggle field for your Fluid field. Call it `{hide_next_block}`, for example.
2. Then, in your template, add this condition: `{if fluid_field:prev_field_name != "hide_next_block"}show your block content{/if}`

This way, if you add this `Hide next block` toggle right before the field you would like to disable - it will be disabled :)

I did this trick for https://vurchel.com and it worked perfectly.

**Reply by pyrobob83** [reply]
Great idea for a workaround, but still would love to see something native that allows control per role.

**Comment by intoeetive** [comment]
Just wanted to say that I like the idea, but we'd need to find a good UX solution for it.

Perhaps a toggle next to field drag icon.

That way, people could hide some of Fluid content instead of removing.

However conditional display (or access) here would be a completely different thing...

**Comment by pyrobob83** [comment]
@intoeetive The simple toggle would be great, but conditional display based on role would be better.  I know there's more to consider such as the UI and how it'd impact caching, but it doesn't feel like it'd be out of reach (granted I'm not a PHP dev).  Has Tom given his input on this?  I'm thinking just as much about improving EE's core as I am about making life easier for myself as a dev, so maybe I'll ping him to give his thoughts on whether it's worth looking further into the role-based part. @TomJaeger

**Comment by pyrobob83** [comment]
@intoeetive @TomJaeger This seems to have gotten forgotten so bumping it again - even just a simple 'disable/enable' toggle would be really useful when using Fluid Fields as a page-builder... there's no way to temporarily unpublish one block without deleting it entirely, which feels like it's a pretty simple and critical feature. Can we resume consideration here?
